src/openrct2/math/Trigonometry.hpp

```diff
@@ -0,0 +1,126 @@
+        { -181, -181 }, // slopes looping down
+        { -221, -128 }, // slopes looping down
+        { -247, -66 },  // slopes looping down
+        { 221, 128 },   // corkscrew up left
+        { 128, 221 },   // corkscrew up left
+        { 0, 256 },     // corkscrew up left
+        { -128, 221 },  // corkscrew up left
+        { -221, 128 },  // corkscrew up left
+        { -221, -128 }, // corkscrew down left
+        { -128, -221 }, // corkscrew down left
+        { 0, -256 },    // corkscrew down left
+        { 128, -221 },  // corkscrew down left
+        { 221, -128 },  // corkscrew down left
+        { 221, 128 },   // corkscrew up right
+        { 128, 221 },   // corkscrew up right
+        { 0, 256 },     // corkscrew up right
+        { -128, 221 },  // corkscrew up right
+        { -221, 128 },  // corkscrew up right
+        { -221, -128 }, // corkscrew down right
+        { -128, -221 }, // corkscrew down right
+        { 0, -256 },    // corkscrew down right
+        { 128, -221 },  // corkscrew down right
+        { 221, 128 },   // corkscrew down right
+        { 256, 0 },     // half helixes
+        { 256, 0 },     // half helixes
+        { 256, 0 },     // half helixes
+        { 256, 0 },     // half helixes
+        { 256, 0 },     // quarter helixes
+        { 256, 0 },     // quarter helixes
+        { 252, 42 },    // diagonal slopes up
+        { 241, 83 },    // diagonal slopes up
+        { 168, 193 },   // diagonal slopes up
+        { 252, -42 },   // diagonal slopes down
+        { 241, -83 },   // diagonal slopes down
+        { 168, -193 },  // diagonal slopes down
+        { 236, -97 },   // inverting transition slopes down
+        { 195, -165 },  // inverting transition slopes down
+        { 134, -217 },  // inverting transition slopes down
+        { 252, 44 },    // spiral lift hill up
+        { 252, -44 }    // spiral lift hill down
+    };
+
+    constexpr int32_t ComputeHorizontalMagnitude(int32_t length, uint8_t pitch)
+    {
+        return (-PitchToDirectionVectorFromGeometry[static_cast<uint8_t>(pitch)].y * length) / 256;
+    }
+
+    constexpr CoordsXY ComputeXYVector(int32_t magnitude, uint8_t yaw)
+    {
+        return (static_cast<CoordsXY>(YawToDirectionVector[yaw]) * magnitude) / 256;
+    }
+    constexpr CoordsXY ComputeXYVector(int32_t length, uint8_t pitch, uint8_t yaw)
+    {
+        return ComputeXYVector(ComputeHorizontalMagnitude(length, pitch), yaw);
+    }
+
+    constexpr CoordsXYZ ComputeSteamOffset(int32_t height, int32_t length, uint8_t pitch, uint8_t yaw)
+    {
+        uint8_t trueYaw = OpenRCT2::Entity::Yaw::YawTo64(yaw);
+        auto offsets = PitchToDirectionVectorFromGeometry[pitch];
+        int32_t projectedRun = (offsets.x * length - offsets.y * height) / 256;
+        int32_t projectedHeight = (offsets.x * height + offsets.y * length) / 256;
+        return { ComputeXYVector(projectedRun, trueYaw), projectedHeight };
+    }
```

Comment:
This probably belongs to SteamParticle.hpp

Reply:
Instead of creating a file just for this one function, could it be in Vehicle.cpp?

Reply:
I didn't realize you deleted it, yes.